src/databricks/labs/ucx/source_code/site_packages.py

```diff
@@ -1,14 +1,67 @@
 from __future__ import annotations
 
 import os
+import subprocess
+import tempfile
+from functools import cached_property
 from pathlib import Path
+
 from databricks.labs.ucx.source_code.files import FileLoader
 from databricks.labs.ucx.source_code.graph import (
+    BaseImportResolver,
     Dependency,
-    SourceContainer,
     DependencyGraph,
     DependencyProblem,
+    MaybeDependency,
+    SourceContainer,
+    WrappingLoader,
 )
+from databricks.labs.ucx.source_code.path_lookup import PathLookup
+
+
+class PipResolver(BaseImportResolver):
+    # TODO: use DistInfoResolver to load wheel/egg/pypi dependencies
+    # TODO: https://github.com/databrickslabs/ucx/issues/1642
+    # TODO: https://github.com/databrickslabs/ucx/issues/1643
+    # TODO: https://github.com/databrickslabs/ucx/issues/1640
+
+    def __init__(
+        self,
+        file_loader: FileLoader,
+        next_resolver: BaseImportResolver | None = None,
+    ):
+        super().__init__(next_resolver)
+        self._file_loader = file_loader
+
+    def with_next_resolver(self, resolver: BaseImportResolver) -> BaseImportResolver:
+        return PipResolver(self._file_loader, resolver)
+
+    def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
+        path_lookup.append_path(self._temporary_virtual_environment)
```

Comment:
@ericvergnaud : Should we update the path look outside of each resolver, i.e. on the top level:

```
lib = self._import_resolver.resolve_import()
self._path_look.append_path(lib.path)
return lib
```